util/chplenv/compile_link_args_utils.py

```diff
@@ -96,8 +96,11 @@ def get_runtime_link_args(runtime_subdir):
         sdk_path = chpl_gpu.get_sdk_path(gpu_type)
         if gpu_type == "nvidia":
             system.append("-L" + os.path.join(sdk_path, "lib64"))
-            system.append("-lcuda")
             system.append("-lcudart")
+            if chpl_platform.is_wsl():
+                # WSL needs to link with libcuda that belongs to the driver hosted in Windows
+                system.append("-L" + os.path.join("/usr", "lib", "wsl", "lib"))
+            system.append("-lcuda")
```

Comment:
the rest of the SDK is in the usual place. I think using this special location as the sdk path breaks it because that's not where `nvcc` gets installed.

truncated output from `printchplenv --internal`
```
  CHPL_GPU_ARCH: sm_60
  CHPL_CUDA_PATH: /usr/local/cuda-11.8
  CHPL_CUDA_LIBDEVICE_PATH: /usr/local/cuda-11.8/nvvm/libdevice/libdevice.10.bc
```

location of `nvcc`:
```shell
$ which nvcc
/usr/local/cuda-11.8/bin/nvcc
```

the contents of the directory where `libcuda` lives:
```shell
$ ls /usr/lib/wsl/lib/
libcudadebugger.so.1  libd3d12core.so  libnvcuvid.so.1        libnvidia-ml.so.1           libnvoptix.so.1
libcuda.so            libd3d12.so      libnvdxdlkernels.so    libnvidia-opticalflow.so    libnvwgf2umx.so
libcuda.so.1          libdxcore.so     libnvidia-encode.so    libnvidia-opticalflow.so.1  nvidia-smi
libcuda.so.1.1        libnvcuvid.so    libnvidia-encode.so.1  libnvoptix_loader.so.1
```